src/fw/drivers/sf32lb52/rng.c

```diff
@@ -15,32 +15,36 @@
  */
 
 #include "drivers/rng.h"
+
 #include "system/passert.h"
 
 #define SF32LB52_COMPATIBLE
 #include "mcu.h"
 
+static RNG_HandleTypeDef s_rng_hdl = {
+    .Instance = hwp_trng,
+};
+
 bool rng_rand(uint32_t *rand_out) {
   PBL_ASSERTN(rand_out);
-  RNG_HandleTypeDef rng_hdl = {0};
   HAL_StatusTypeDef status;
 
-  rng_hdl.Instance = hwp_trng;
-
-  status = HAL_RNG_Init(&rng_hdl);
+  HAL_RCC_EnableModule(RCC_MOD_TRNG);
+  status = HAL_RNG_Init(&s_rng_hdl);
   if (status != HAL_OK) {
     PBL_LOG(LOG_LEVEL_ERROR, "rng_rand init fail!");
     return false;
   }
 
-  status = HAL_RNG_Generate(&rng_hdl, rand_out, 1);
+  status = HAL_RNG_Generate(&s_rng_hdl, rand_out, 1);
+
+  HAL_RNG_DeInit(&s_rng_hdl);
+  HAL_RCC_DisableModule(RCC_MOD_TRNG);
```

Comment:
question: is RNG state retained if RCC is disabled? I assume if RCC is disabled, RNG will be gated so it won't consume? If that's the case, we could just initialize once?

Reply:
RCC enable is default state, it work seperately with HAL_RNG_* API. And `HAL_RNG_Init` looks have no effect with real register. `HAL_RNG_Generate` start the enegine, `HAL_RNG_DeInit` stop the enegine.

I add the clear 'stop' flag in `__HAL_RNG_ENABLE`, because there have a system halt problem if call `HAL_RNG_Init`, `HAL_RNG_Generate` and `HAL_RNG_DeInit` each time.